src/SIL.Harmony.Tests/DataModelPerformanceTests.cs

```diff
@@ -0,0 +1,233 @@
+using System.Diagnostics;
+using System.Text;
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Columns;
+using BenchmarkDotNet.Configs;
+using BenchmarkDotNet.Engines;
+using BenchmarkDotNet.Loggers;
+using BenchmarkDotNet.Running;
+using JetBrains.Profiler.SelfApi;
+using Microsoft.Data.Sqlite;
+using SIL.Harmony.Changes;
+using SIL.Harmony.Core;
+using SIL.Harmony.Db;
+using SIL.Harmony.Sample.Models;
+using Xunit.Abstractions;
+
+namespace SIL.Harmony.Tests;
+
+[Trait("Category", "Performance")]
+public class DataModelPerformanceTests(ITestOutputHelper output)
+{
+    [Fact]
+    public void AddingChangePerformance()
+    {
+        var summary =
+            BenchmarkRunner.Run<DataModelPerformanceBenchmarks>(
+                ManualConfig.CreateEmpty()
+                    .AddColumnProvider(DefaultColumnProviders.Instance)
+                    .AddLogger(new XUnitBenchmarkLogger(output))
+            );
+        foreach (var benchmarkCase in summary.BenchmarksCases.Where(b => !summary.IsBaseline(b)))
+        {
+            var ratio = double.Parse(BaselineRatioColumn.RatioMean.GetValue(summary, benchmarkCase));
+            //for now it just makes sure that no case is worse that 7x, this is based on the 10_000 test being 5 times worse.
+            //it would be better to have this scale off the number of changes
+            ratio.Should().BeInRange(0, 7, "performance should not get worse, benchmark " + benchmarkCase.DisplayInfo);
+        }
+    }
+    
+    //enable this to profile tests
+    private static readonly bool trace = (Environment.GetEnvironmentVariable("DOTNET_TRACE") ?? "false") != "false";
+    private async Task StartTrace()
+    {
+        if (!trace) return;
+        await DotTrace.InitAsync();
+        // config that sets the save directory
+        var config = new DotTrace.Config();
+        var dirPath = Path.Combine(Path.GetTempPath(), "harmony-perf");
+        Directory.CreateDirectory(dirPath);
+        config.SaveToDir(dirPath);
+        DotTrace.Attach(config);
+        DotTrace.StartCollectingData();
+    }
+    
+    private void StopTrace()
+    {
+        if (!trace) return;
+        DotTrace.SaveData();
+        DotTrace.Detach();
+    }
+    
+    private static async Task<TimeSpan> MeasureTime(Func<Task> action, int iterations = 10)
+    {
+        var total = TimeSpan.Zero;
+        for (var i = 0; i < iterations; i++)
+        {
+            var start = Stopwatch.GetTimestamp();
+            await action();
+            total += Stopwatch.GetElapsedTime(start);
+        }
+        return total / iterations;
+    }
+
+    [Fact]
+    public async Task SimpleAddChangePerformanceTest()
```

Comment:
This test doesn't use any of the benchmarking code/libraries you added, right?
What's the difference/why do we have both this and the benchmarks?

Reply:
That's right. The main reason is that this one is fast but less accurate, and the other one is slower but more accurate. 

I made this one for iteration on performance because the other was too slow